File: LeetCode/1863. Sum of All Subset XOR Totals/1863.py
class Solution(object):
    def subsetXORSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # 'bits' will store the bitwise OR of all numbers
        bits = 0
        for n in nums:
            # Accumulate set bits across all numbers
            bits = bits | n        
            # Return the total sum by multiplying 'bits' by 2 raised to the power of (length of 'nums' - 1)
        return bits * 2**(len(nums)-1)

File: LeetCode/1863. Sum of All Subset XOR Totals/1863.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        int n = nums.size();
        int subsetNum =1<<n;
        int total=0;
        
        //generate each subset one by one
        for(int i=0;i<subsetNum;i++){
            int xorSum=0;
            for(int j=0;j<n;j++){
                if(i&(1<<j)){
                    xorSum^=nums[j];
                }
            }
            total+=xorSum;
        }
        return total;
        
    }
};